pytorch_lightning/trainer/trainer.py

```diff
@@ -422,6 +422,14 @@ def __init__(
         self.optimizer_connector.on_trainer_init()
 
         # init data flags
+        if prepare_data_per_node is None:
+            prepare_data_per_node = True
+        else:
+            rank_zero_warn(
```

Comment:
use `rank_zero_deprecated` for deprecation warnings

Reply:
gotchat.

Reply:
you may also want to move this deprecation into the `data_connector.on_trainer_init`